src/rust/engine/src/externs/interface.rs

```diff
@@ -943,106 +941,116 @@ async fn workunit_to_py_value(
   core: &Arc<Core>,
   session: &Session,
 ) -> CPyResult<Value> {
-  use std::time::UNIX_EPOCH;
+  let mut dict_entries = {
+    let gil = Python::acquire_gil();
+    let py = gil.python();
+    let mut dict_entries = vec![
```

Comment:
Since this `Vec` will likely have elements pushed, maybe allocate via `Vec::with_capacity(N)` for some N?

Reply:
True, although probably better as a followup. It looks like there will be at least 10-12 extra elements added